File: LeetCode_Record/2293.极大极小游戏.java
class Solution {
    public int minMaxGame(int[] nums) {
        int n = nums.length;
        while(n > 1){
            for(int i = 0; i < n/2; i++){
                if(i % 2 == 1){
                    nums[i] = Math.max(nums[i * 2], nums[i * 2 + 1]);
                }else{
                    nums[i] = Math.min(nums[i * 2], nums[i * 2 + 1]);
                }
            }
            n /= 2;
        }
        return nums[0];
    }
}

File: LeetCode_Record/2293.极大极小游戏.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int minMaxGame(vector<int>& nums) {
        int n = nums.size();
        while(n > 1){
            for(int i = 0; i < n/2; i++){
                if(i % 2)
                    nums[i] = max(nums[i * 2], nums[i * 2 + 1]);
                else
                    nums[i] = min(nums[i * 2], nums[i * 2 + 1]);
            }
            n /= 2;
        }
        return nums[0];
    }
};